diff --git a/dc-cudami-editor/src/components/IdentifiableForm.jsx b/dc-cudami-editor/src/components/IdentifiableForm.jsx
index 7171bfd1..fba6e21e 100644
--- a/dc-cudami-editor/src/components/IdentifiableForm.jsx
+++ b/dc-cudami-editor/src/components/IdentifiableForm.jsx
@@ -25,6 +25,7 @@ import IframeAdderModal from './modals/IframeAdderModal'
 import ImageAdderModal from './modals/ImageAdderModal'
 import LanguageAdderModal from './modals/LanguageAdderModal'
 import LinkAdderModal from './modals/LinkAdderModal'
+import PreviewImageAdderModal from './modals/PreviewImageAdderModal'
 import TableAdderModal from './modals/TableAdderModal'
 import initI18n from '../i18n'
 import '../polyfills'
@@ -51,6 +52,7 @@ class IdentifiableForm extends Component {
         imageAdder: false,
         languageAdder: false,
         linkAdder: false,
+        previewImageAdder: false,
         tableAdder: false,
       },
     }
@@ -274,6 +276,14 @@ class IdentifiableForm extends Component {
           isOpen={this.state.modalsOpen.linkAdder}
           onToggle={() => this.toggleModal('linkAdder')}
         />
+        <PreviewImageAdderModal
+          activeLanguage={this.state.activeLanguage}
+          apiContextPath={this.props.apiContextPath}
+          debug={this.props.debug}
+          defaultLanguage={this.state.defaultLanguage}
+          isOpen={this.state.modalsOpen.previewImageAdder}
+          onToggle={() => this.toggleModal('previewImageAdder')}
+        />
         <TableAdderModal
           isOpen={this.state.modalsOpen.tableAdder}
           onToggle={() => this.toggleModal('tableAdder')}
diff --git a/dc-cudami-editor/src/components/Teaser.jsx b/dc-cudami-editor/src/components/Teaser.jsx
index c2314fe5..1bf4f0d4 100644
--- a/dc-cudami-editor/src/components/Teaser.jsx
+++ b/dc-cudami-editor/src/components/Teaser.jsx
@@ -1,28 +1,49 @@
 import React from 'react'
-import {Card, CardBody} from 'reactstrap'
+import {Card, CardBody, Col, Row} from 'reactstrap'
 
 import FormEditor from './FormEditor'
 import FormLabelInput from './FormLabelInput'
+import PreviewImage from './PreviewImage'
 
-const Teaser = ({description, label, language, onUpdate}) => {
+const Teaser = ({
+  description,
+  label,
+  language,
+  onUpdate,
+  previewImage,
+  previewImageRenderingHints,
+  updatePreviewImage,
+}) => {
   return (
     <Card className="bg-light">
       <CardBody>
-        <FormLabelInput
-          label={label}
-          language={language}
-          onUpdate={(label) => {
-            onUpdate('label', label)
-          }}
-        />
-        <FormEditor
-          document={description}
-          onUpdate={(document) => {
-            onUpdate('description', document)
-          }}
-          restrictedMenu={true}
-          type="description"
-        />
+        <Row>
+          <Col sm="2">
+            <PreviewImage
+              language={language}
+              previewImage={previewImage}
+              previewImageRenderingHints={previewImageRenderingHints}
+              onUpdate={updatePreviewImage}
+            />
+          </Col>
+          <Col sm="10">
+            <FormLabelInput
+              label={label}
+              language={language}
+              onUpdate={(label) => {
+                onUpdate('label', label)
+              }}
+            />
+            <FormEditor
+              document={description}
+              onUpdate={(document) => {
+                onUpdate('description', document)
+              }}
+              restrictedMenu={true}
+              type="description"
+            />
+          </Col>
+        </Row>
       </CardBody>
     </Card>
   )
diff --git a/dc-cudami-editor/src/components/WebpageForm.jsx b/dc-cudami-editor/src/components/WebpageForm.jsx
index 45e21b6f..3524a40a 100644
--- a/dc-cudami-editor/src/components/WebpageForm.jsx
+++ b/dc-cudami-editor/src/components/WebpageForm.jsx
@@ -86,6 +86,11 @@ const WebpageForm = ({
                       },
                     })
                   }
+                  previewImage={identifiable.previewImage}
+                  previewImageRenderingHints={
+                    identifiable.previewImageRenderingHints
+                  }
+                  updatePreviewImage={onUpdate}
                 />
                 <Card className="border-top-0">
                   <CardBody>
